### DISCUSSION TITLE
Knock alert

### DISCUSSION THREAD

**Discussion by gabber366** [dgreif/ring#582] [Ideas]
Hello! I love that you have brought ring to Homekit through homebridge! This has been wonderful! I have a Ring peephole cam and I was wondering if there is anyway to add the Knock detection feature to the software so it pings my HomePods and shows up on the Apple TVs?

**Comment by dgreif** [comment]
Yep, detecting the knock should definitely be possible.  I've done some initial research as part of #514 and plan get it added sometime soon.  It will probably be best to add it as a new option like `"treatKnockAsDing": true`.  I'll ping you on that issue when it's ready for some testing

**Comment by galoiring** [comment]
**edit**
great update work perfect !


there is any way I can help advance this ? (c \ python knowledge )
